src/js/media-controller.js

```diff
@@ -38,357 +52,724 @@ const ButtonPressedKeys = ['ArrowLeft', 'ArrowRight', 'Enter', ' ', 'f', 'm', 'k
 const DEFAULT_SEEK_OFFSET = 10;
 const DEFAULT_TIME = 0;
 
-/**
- * Media Controller should not mimic the HTMLMediaElement API.
- * @see https://github.com/muxinc/media-chrome/pull/182#issuecomment-1067370339
- */
-class MediaController extends MediaContainer {
-  static get observedAttributes() {
-    return super.observedAttributes.concat('nohotkeys', 'hotkeys', 'default-stream-type');
-  }
+const MediaUIStates = {
+  MEDIA_PAUSED: {
+    get: function(controller){
+      const { media } = controller;
+
+      return media ? media.paused : true;
+    },
+    mediaEvents: ['play', 'pause', 'emptied']
+  }, 
+  MEDIA_HAS_PLAYED: {
+    // We want to let the user know that the media started playing at any point (`media-has-played`).
+    // Since these propagators are all called when boostrapping state, let's verify this is
+    // a real playing event by checking that 1) there's media and 2) it isn't currently paused.
+    get: function(controller) {
+      const { media } = controller;
+
+      if (!media) return false;
+
+      // TODO: Would seem to leave room for issue if this is called
+      // after a video has played but is currently paused.
```

Comment:
@cjpillsbury I'm not gonna try to fix this as part of this PR, but can you confirm this is a potential issue? e.g.
* Load a player
* press play
* press pause
* Add a poster element

The poster element would get has_played = false

Reply:
Let's make sure we don't conflate "has loaded" with "has played". I believe this is an example where we need to "locally store and reset state", since it's not something that can be deterministically derived from the media element and instead is "historic" in nature. lmk if you want a "reliable but maybe a quick hack/not final form" solution. Also, glad you called this out, as this is likely broken. Making me a bit nervous on not having a simple diff to look at for these cases... 😅

Reply:
@heff since this appears to be a regression edge case, should we treat this as out of scope for this initial pass/PR?